proto/api_spec/method.proto

```diff
@@ -150,11 +150,40 @@ message HTTPHeader {
 
 message HTTPAuth {
   enum HTTPAuthType {
+    //
+    // Standard types (IETF/IANA)
+    // https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml
+    //
     UNKNOWN = 0;
     BASIC = 1;
     BEARER = 2;
+    DIGEST = 3; 
+    MUTUAL = 4;
+    OAUTH = 5;
+    VAPID = 6;
+    SCRAM_SHA_1 = 7;
+    SCRAM_SHA_256 = 8;
+    NEGOTIATE = 9;
+    HOBA = 10;
+
+    // An application-specific authorization header, such as
+    // Github webhook's X-Hub-Signature-256 or X-Hub-Signature,
+    // *not* carried in "Authorization".
+    PROPRIETARY_HEADER = 11;
```

Comment:
I think consumers of the IR currently assume that these authorization types are associated with the `Authorization` header. We'll need to weed out all the places that make this assumption.

Reply:
The conversion to/from OpenAPI2/3 will need to be updated, as well as the translation of the IR -> frontend data structures.  Maybe the visitors as well, @liujed?  Seems doable though.

Reply:
As I recall, parts of the GitHub comment generator will need to be updated too.